### PR TITLE
Fabo/select signer per network

### PR DESCRIPTION
**Pull request by faboweb**
Select signing method based on network type

Merge https://github.com/luniehq/lunie-api/pull/378 first

### CODE REVIEW

File: src/ActionModal/components/ActionModal.vue [hunk 1]
```diff
@@ -486,6 +486,7 @@ export default {
         url: this.network.api_url,
         networkId: this.network.id,
         chainId: this.network.chain_id,
+        networkType: this.network.network_type,
```
Comment (Bitcoinera) [inline src/ActionModal/components/ActionModal.vue]: How did you get this to work without having `network_type` in the Network schema? (I just added it to your PR)
O.o

File: src/ActionModal/utils/ActionManager.js [hunk 2]
```diff
@@ -192,7 +193,7 @@ export default class ActionManager {
       txMessages.push(txMessage)
     }
 
-    const createSignedTransaction = await getTransactionSigner(context)
+    const createSignedTransaction = await signedTransactionCreator(context)
```
Comment (Bitcoinera) [inline src/ActionModal/utils/ActionManager.js]: I think the naming of this function can be improved. Why not `getSignedByTransactionCreator` or `signByTransactionCreator`? I think both are more descriptive names

File: src/ActionModal/utils/signer.js [hunk 3]
```diff
@@ -11,57 +11,75 @@ export async function cancelSign(submitType = "", { address, network }) {
...
+    )
+
+    return {
+      signature,
+      publicKey: Buffer.from(wallet.publicKey, "hex")
+    }
+  }
+}
+async function getCosmosLedgerSigner(config) {
+  // TODO show which properties of config are actually needed
```
Comment (Bitcoinera) [inline src/ActionModal/utils/signer.js]: What properties of config? Why is this now needed?

File: src/ActionModal/utils/signer.js [hunk 4]
```diff
@@ -11,57 +11,75 @@ export async function cancelSign(submitType = "", { address, network }) {
...
+          "Something went wrong connecting to your Ledger. Please refresh your page and try again."
+        )
       }
+      throw err
     }
-  } else if (submitType === `extension`) {
-    return signMessage => {
-      return signWithExtension(signMessage, address, network)
+
+    // cleanup. if we leave this open, the next connection will brake for HID
```
Comment (Bitcoinera) [inline src/ActionModal/utils/signer.js]:
```suggestion
    // cleanup. if we leave this open, the next connection will break for HID
```

File: src/components/common/TmSessionExtension.vue [hunk 5]
```diff
@@ -62,15 +62,6 @@ export default {
...
   },
-  async created() {
-    if (this.$route.params.address && this.$route.params.network) {
-      await this.signIn({
-        address: this.$route.params.address,
-        network: this.$route.params.network
-      })
-      this.$router.push("/portfolio")
-    }
-  },
```
Comment (Bitcoinera) [inline src/components/common/TmSessionExtension.vue]: Deleting this is not related to these PR changes no? Or it is?

File: src/routes.js [hunk 6]
```diff
@@ -316,3 +315,27 @@ export default (apollo, store) => {
...
+// handle direct sign in from the extension via deeplink
+export async function extensionSignIn({ to, next }, apollo, store) {
+  const {
+    data: { network }
+  } = await apollo.query({
+    query: gql`
+      query Network {
+        network(id: "${to.params.network}") {
+          id
+          slug
```
Comment (Bitcoinera) [inline src/routes.js]: Why are we not sending here `network_type` as well?

File: tests/unit/specs/components/ActionModal/utils/ActionManager.spec.js [hunk 7]
```diff
@@ -29,8 +29,8 @@ const mockMsgWithdraw = jest.fn(() => ({
   send: () => ({ included: () => async () => true })
 }))
 
-const mockGetTransactionSigner = jest.fn(() => {
-  console.log("mockGetTransactionSigner executed")
+const mocksignedTransactionCreator = jest.fn(() => {
```
Comment (Bitcoinera) [inline tests/unit/specs/components/ActionModal/utils/ActionManager.spec.js]: I advice changing the function name

**Comment by Bitcoinera** [comment]
It looks quite good though. Going through with the debugger and everything seems to be working as expected 💃

**Comment by Bitcoinera** [comment]
e2e tests keep failing, but if we are going for #3629 it is not worth it to fix them for this one

File: src/ActionModal/utils/ActionManager.js (hunk 2 above)
Comment (faboweb) [reply src/ActionModal/utils/ActionManager.js]: Why are those `signedBy`?

File: src/ActionModal/utils/signer.js (hunk 3 above)
Comment (faboweb) [reply src/ActionModal/utils/signer.js]: This was in the code before. This is just a refactor. `config` is passed through to the library. It would be nice to show here which properties of `config.js` are actually needed. This would help with search and replace or when we want to delete things from config.js.

File: src/components/common/TmSessionExtension.vue (hunk 5 above)
Comment (faboweb) [reply src/components/common/TmSessionExtension.vue]: Nop, sorry might have slipped in.

File: src/routes.js (hunk 6 above)
Comment (faboweb) [reply src/routes.js]: not needed here, is it?

File: tests/unit/specs/components/ActionModal/utils/ActionManager.spec.js (hunk 7 above)
Comment (faboweb) [reply tests/unit/specs/components/ActionModal/utils/ActionManager.spec.js]: why?

File: src/routes.js (hunk 6 above)
Comment (Bitcoinera) [reply src/routes.js]: to tell the extension the network type that need to sign on it, no? I guess in the extension we will also introduce this `getCosmosSigner` & `getWhateverNetworkSigner` logic? or I got that wrong?

File: src/ActionModal/utils/signer.js (hunk 3 above)
Comment (Bitcoinera) [reply src/ActionModal/utils/signer.js]: Oh right. Thank you so much for the clarification

File: src/ActionModal/components/ActionModal.vue (hunk 1 above)
Comment (Bitcoinera) [reply src/ActionModal/components/ActionModal.vue]: Magic xD

**Comment by Bitcoinera** [comment]
The e2e tests seem to be failing in exactly the same thing that @iambeone  had problems with. The `:nth-of-type(2)` element being not visible.

File: src/components/common/TmBalance.vue [hunk 8]
```diff
@@ -109,6 +109,7 @@
                     <span
                       v-if="
                         isMultiDenomNetwork &&
+                          stakingBalance.fiatValue &&
```
Comment (Bitcoinera) [inline src/components/common/TmBalance.vue]: hey, this looks familiar to me! hehe

File: src/components/common/TmBalance.vue [hunk 9]
```diff
@@ -166,7 +167,11 @@
                     </h2>
                   </div>
                   <div
-                    v-if="balance.fiatValue.amount > 0 && preferredCurrency"
+                    v-if="
+                      balance.fiatValue &&
```
Comment (Bitcoinera) [inline src/components/common/TmBalance.vue]: same 😛
